targets/TARGET_ONSEMI/TARGET_NCS36510/gpio_api.c

```diff
@@ -143,7 +143,9 @@ void gpio_mode(gpio_t *obj, PinMode mode)
 void gpio_dir(gpio_t *obj, PinDirection direction)
 {
     /* Enable the GPIO clock */
-    CLOCK_ENABLE(CLOCK_GPIO);
+    if(!CLOCK_IS_ENABLED(CLOCK_GPIO)) {
+        CLOCK_ENABLE(CLOCK_GPIO);
+    }
```

Comment:
I'd like to ask @0xc0170 whether he really thinks this is worthwhile. Reading the hardware to find out if it's worth writing to the hardware? More code, and probably no faster - usually 1 read instead of 1 write.